lib/rspec/bash/util/call_conf_argument_list_matcher.rb

```diff
@@ -0,0 +1,47 @@
+require 'rspec/mocks/argument_list_matcher'
+include RSpec::Mocks
+include RSpec::Mocks::ArgumentMatchers
+
+module Rspec
+  module Bash
+    module Util
+      class CallConfArgumentListMatcher < ArgumentListMatcher
+        alias parent_args_match? args_match?
+        alias parent_initialize initialize
+
+        def initialize(call_conf_list)
+          @expected_call_conf_list = call_conf_list
+        end
+
+        def args_match?(*call_arguments)
+          !get_call_conf_matches(*call_arguments).empty?
+        end
+
+        def get_best_call_conf(*call_arguments)
+          get_call_conf_matches(*call_arguments).sort_by do |call_conf|
+            [
+              call_conf[:args].length
+            ]
+          end.last || {}
+        end
+
+        def get_call_conf_matches(*call_arguments)
+          @expected_call_conf_list.select do |expected_call_conf|
+            @expected_args = remap_argument_matchers(expected_call_conf[:args])
+            parent_args_match?(*call_arguments)
```

Comment:
I can't remember why I had to do this. I think it was because ruby's super method calling is goofy. Anyways, How this works is:
- at top of class, alias the parent instance method that you want to call (args_match?) to a more expressive name (parent_args_match?)
- When you want to call that parent instance method, call parent_args_match? It's like super.args_match? in Java. The naming could be changed to make it more clear if you want.